### PR TITLE
Add a ReAct agent and an example conversation with this agent. 

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

- Add ReAct agent
- Add an example for conversation with react agent

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: examples/conversation_with_react_agent/code/conversation_with_react_agent.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    verbose=True,
+    tools=tools,
+)
+user = UserAgent(name="User")
+
+# Build
+x = None
+while True:
+    x = user(x)
+    if x is None:
```
Comment (ZiTao-Li) [inline examples/conversation_with_react_agent/code/conversation_with_react_agent.py:85]: Does UserAgent ever return None? Maybe need to check content.

**Review by ZiTao-Li — COMMENTED** [review]
See a comment in the example. LGTM in general as a preliminary example for ReAct. We can iterate to new versions as feedbacks come. At this point, we may need to consider the following points:
1. Do we need to modify agentscope.service.execute_python_code to accommodate the need of ReAct or tools agent?
2. Do we consider returning a more meaningful message in the reply function?

File: examples/conversation_with_react_agent/code/conversation_with_react_agent.py (hunk 1 above)
Comment (DavdGao) [reply examples/conversation_with_react_agent/code/conversation_with_react_agent.py]: It's a typo, and fixed in new commit.

**Comment by DavdGao** [comment]
> See a comment in the example. LGTM in general as a preliminary example for ReAct. We can iterate to new versions as feedbacks come. At this point, we may need to consider the following points:
> 
> 1. Do we need to modify agentscope.service.execute_python_code to accommodate the need of ReAct or tools agent?
> 2. Do we consider returning a more meaningful message in the reply function?

1. We can decouple the current built-in `execute_python_code` into `execute_python_in_docker`, `execute_python_in_process` and `execute_python` functions. Or leaving it to be fixed in the future  (related issue #97 ). @rayrayraykk @ZiTao-Li What do you think?

**Comment by rayrayraykk** [comment]
> > See a comment in the example. LGTM in general as a preliminary example for ReAct. We can iterate to new versions as feedbacks come. At this point, we may need to consider the following points:
> > 
> > 1. Do we need to modify agentscope.service.execute_python_code to accommodate the need of ReAct or tools agent?
> > 2. Do we consider returning a more meaningful message in the reply function?
> 
> 1. We can decouple the current built-in `execute_python_code` into `execute_python_in_docker`, `execute_python_in_process` and `execute_python` functions. Or leaving it to be fixed in the future  (related issue [[Feature]: execute python code in thread #97](https://github.com/modelscope/agentscope/issues/97) ). @rayrayraykk @ZiTao-Li What do you think?

Agree to decouple the `execute_python_code`, since the Agent do not need to feel the differences of each running env and it is actually controlled by programmers.

**Comment by ZiTao-Li** [comment]
> > > See a comment in the example. LGTM in general as a preliminary example for ReAct. We can iterate to new versions as feedbacks come. At this point, we may need to consider the following points:
> > > 
> > > 1. Do we need to modify agentscope.service.execute_python_code to accommodate the need of ReAct or tools agent?
> > > 2. Do we consider returning a more meaningful message in the reply function?
> > 
> > 
> > 
> > 1. We can decouple the current built-in `execute_python_code` into `execute_python_in_docker`, `execute_python_in_process` and `execute_python` functions. Or leaving it to be fixed in the future  (related issue [[Feature]: execute python code in thread #97](https://github.com/modelscope/agentscope/issues/97) ). @rayrayraykk @ZiTao-Li What do you think?
> 
> Agree to decouple the `execute_python_code`, since the Agent do not need to feel the differences of each running env and it is actually controlled by programmers.

Agree. It is good for this version; the decoupling can be done in the future.

File: src/agentscope/agents/react_agent.py
Comment (pan-x-c) [inline src/agentscope/agents/react_agent.py:1]: add an item to sphinx doc index file (`docs/sphinx_doc/en/source/agentscope.agents.rst` and `docs/sphinx_doc/zh_CN/source/agentscope.agents.rst`)

Comment (DavdGao) [reply src/agentscope/agents/react_agent.py:1]: Done

**Review by pan-x-c — APPROVED** [review]
LGTM

**Pull request merged**